### DISCUSSION TITLE
Link Temperatures for AC Partner devices?

### DISCUSSION THREAD

**Discussion by TJT-JTJ** [merdok/homebridge-miot#347] [Q&A]
I have been tempted to buy the new Aqara P3... but it's fairly expensive and I already have several working acpartner devices perfectly working with homebridge.

Though I'd like to improve it a bit. The temperature reported within the Home app is the temperature the AC is currently set to, not the 'correct' temperature. The Aqara P3 overcomes this by forcing you to use a separate temperature device to make the Home app display correctly. 

What I'm wondering is if there's a way to link the temperature from another device and have homebridge use that as the temperature. For example I've got an air purifier which reports the temperature constantly and is also connected via homebridge, so the data is all there, I'm just not sure how or if it's possible to tell homebridge to use the data from another device. 

Thanks for your suggestions.

**Comment by merdok** [comment]
I do not quite understand what you are trying to accomplish. You want to show the temperature reported by the air purifier on the AC? Why? You already have that displayed in home through the air purifier....

**Reply by TJT-JTJ** [reply]
The way it seems to work for native devices is that the heater/cooler devices have a set temperature and the actual temperature - so it will show the actual temperature and say something like 'cooling to xxº'. 

![image](https://user-images.githubusercontent.com/63647882/193411524-ea66ec91-0f26-4e8e-a4b8-a519b38c64fb.png)

I found an example online, the reported temperature for the device is the actual temperature, and then inside it has your set temperature. Because of the way it is currently set up, it says my living room is constantly 25º because that's the set temperature of the AC (even when it is turned off).

<img width="69" alt="Screenshot 2022-10-01 at 21 22 46" src="https://user-images.githubusercontent.com/63647882/193411591-ffe807de-c4e7-4bfb-a91d-aaa4b41c8b8f.png">

Better view here - the temperature is normally actual, not the set one.

Edit: The reason I mentioned the Aqara P3 set up is because they've combined two devices to make it compatible with Home, the actual temperature comes from a separate zigbee temp sensor and the set temperature comes from whatever you set the AC to be at.

**Reply by merdok** [reply]
To display the air purifier temperature is not possible, that would require some very specific changes to be done to make that work. But i guess the idea is if you want to have automations based on that they you monitor the air purifier temperature and control the AC device in those automations, and for that no device mixing is necessary.

**Reply by TJT-JTJ** [reply]
The main issue really is that the AC set temperature is listed as the temperature of the room despite there being multiple temperature devices, so I'm guessing that can't be resolved easily.

**Reply by merdok** [reply]
As i might agree that this is not that optimal, but as far as i remember some people requested it to be like that 😅
